tests/build_/test_arrangements.py

```diff
@@ -0,0 +1,246 @@
+
+
+class TestArrangementV1(ArrangementBase):
+    """
+    This class tests support for the oldest supported arrangement
+    version, 1.
+
+    NOTE! When removing this test class, *make sure* that any methods
+    it provides for the test class for the next oldest supported
+    arrangement version are copied across to that test class.
+    """
+
+    ARRANGEMENT_VERSION = 1
+
+    COMMON_PARAMS = {
+        'git_uri': TEST_GIT_URI,
+        'git_ref': TEST_GIT_REF,
+        'git_branch': TEST_GIT_BRANCH,
+        'user': 'john-foo',
+        'component': TEST_COMPONENT,
+    }
+
+    ORCHESTRATOR_ADD_PARAMS = {
+        'platforms': ['x86_64'],
+    }
+
+    WORKER_ADD_PARAMS = {
+        'platform': 'x86_64',
+        'release': 1,
+    }
+
+    @pytest.mark.parametrize('base_image', [  # noqa:F811
+        'koji/image-build',
+        'foo',
+    ])
+    @pytest.mark.parametrize('scratch', [False, True])
+    def test_add_filesystem_in_orchestrator(self, osbs, scratch, base_image):
+        """
+        Orchestrator builds should not run add_filesystem
+        """
+        additional_params = {
+            'base_image': base_image,
+            'yum_repourls': ['https://example.com/my.repo'],
+        }
+        if scratch:
+            additional_params['scratch'] = True
+
+        (_, build_json) = self.get_orchestrator_build_request(osbs,
+                                                              additional_params)
+
+        self.assert_plugin_not_present(build_json,
+                                       'prebuild_plugins',
+                                       'add_filesystem')
+
+    @pytest.mark.parametrize('scratch', [False, True])  # noqa:F811
+    @pytest.mark.parametrize('base_image, expect_plugin', [
+        ('koji/image-build', True),
+        ('foo', False)
+    ])
+    def test_add_filesystem_in_worker(self, osbs, base_image, scratch,
+                                      expect_plugin):
+        additional_params = {
+            'base_image': base_image,
+            'yum_repourls': ['https://example.com/my.repo'],
+        }
+        if scratch:
+            additional_params['scratch'] = True
+        params, build_json = self.get_worker_build_request(osbs,
+                                                           additional_params)
+        plugins = get_plugins_from_build_json(build_json)
+
+        if not expect_plugin:
+            with pytest.raises(NoSuchPluginException):
+                get_plugin(plugins, 'prebuild_plugins', 'add_filesystem')
+        else:
+            args = plugin_value_get(plugins, 'prebuild_plugins',
+                                    'add_filesystem', 'args')
+
+            allowed_args = set([
+                'koji_hub',
+                'koji_proxyuser',
+                'repos',
+            ])
+            assert set(args.keys()) <= allowed_args
+            assert 'koji_hub' in args
+            assert args['repos'] == params['yum_repourls']
+
+    # ...
```

Comment:
"Needs more tests"

This PR is basically the framework, not a complete test suite for the arrangement version 1.